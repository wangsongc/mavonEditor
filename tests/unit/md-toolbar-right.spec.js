import { shallowMount } from '@vue/test-utils'
import MdToolbarRight from '@/components/md-toolbar-right.vue'
import { CONFIG } from '@/lib/config.js'

/*eslint-disable */
const d_words_zh_cn = CONFIG[`words_zh-CN`]

const factory = (propsData) => {
  return shallowMount(MdToolbarRight, {
    propsData: {
      ...propsData
    }
  })
}
const props = {
  s_subfield: true,
  toolbars: {
    'navigation': true,
    'subfield': true,
    'fullscreen': true,
    'readmodel': true,
    'htmlcode': true,
    'help': true,
    'preview': true
  },
  s_preview_switch: true,
  s_fullScreen: true,
  s_html_code: true,
  s_navigation: true,
  d_words: d_words_zh_cn
}

describe('通过按钮点击调用$clicks方法，检查emit事件', () => {
  it('点击工具栏触发事件', () => {
    //挂载左工具栏组件，传入props
    const wrapper = new factory(props)
    //检查字体加粗元素是否渲染
    expect(wrapper.find('.op-icon.fa.fa-mavon-columns').exists()).toBe(true)
    //点击字体加粗按钮触发$clicks
    wrapper.find('.op-icon.fa.fa-mavon-columns').trigger('click')
    //检查点击事件触发的emit事件
    expect(wrapper.emitted().toolbar_right_click[0]).toEqual(['subfield'])
  })
})
describe('Snapshot', () => {
  it('Snapshot', () => {
    const wrapper = new factory(props)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
