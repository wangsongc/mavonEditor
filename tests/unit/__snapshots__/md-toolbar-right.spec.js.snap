// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Snapshot 1`] = `<div class="v-right-item"> <button type="button" aria-hidden="true" title="开启标题导航 (F8)" class="op-icon fa fa-mavon-bars" style="display: none;"></button> <button type="button" aria-hidden="true" title="关闭标题导航 (F8)" class="op-icon fa fa-mavon-bars selected"></button> <button type="button" aria-hidden="true" title="编辑 (F9)" class="op-icon fa fa-mavon-eye-slash selected"></button> <button type="button" aria-hidden="true" title="预览 (F9)" class="op-icon fa fa-mavon-eye" style="display: none;"></button> <button type="button" title="全屏编辑 (F10)" aria-hidden="true" class="op-icon fa fa-mavon-arrows-alt" style="display: none;"></button> <button type="button" title="退出全屏 (F10)" aria-hidden="true" class="op-icon fa fa-mavon-compress selected"></button> <button type="button" aria-hidden="true" title="沉浸式阅读 (F11)" class="op-icon fa fa-mavon-window-maximize"></button> <button type="button" aria-hidden="true" title="单栏 (F12)" class="op-icon fa fa-mavon-columns selected"></button> <span class="op-icon-divider"></span> <button type="button" title="查看html文本" aria-hidden="true" class="op-icon fa fa-mavon-code" style="display: none;"></button> <button type="button" title="返回markdown文本" aria-hidden="true" class="op-icon fa fa-mavon-code selected"></button> <button type="button" title="markdown语法帮助" aria-hidden="true" class="op-icon fa fa-mavon-question-circle" style="font-size: 17px; padding: 5px 6px 5px 3px;"></button> </div>`;
